### PR TITLE
Fix usage of relative paths or WatchedDir input paths

### LINKED ISSUE #52: v4.1.0 does not handle relative inputs
We just tried to update from v4.0.0 to v4.1.0 and for some reason the new version seems to cause build failures:

```
ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'

=================================================================================

ENV Summary:

  TIME: Thu May 16 2019 20:52:51 GMT+0000 (Coordinated Universal Time)
  TITLE: ember
  ARGV:
  - /usr/local/bin/node
  - /builds/qonto/qonto-js/node_modules/.bin/ember
  - exam
  - --split=8
  - --parallel
  EXEC_PATH: /usr/local/bin/node
  TMPDIR: /tmp
  SHELL: /bin/bash
  PATH:
  - /tmp/yarn--1558039840911-0.3519560502598835
  - /builds/qonto/qonto-js/node_modules/.bin
  - /usr/local/share/.config/yarn/link/node_modules/.bin
  - /builds/qonto/qonto-js/node_modules/.bin
  - /usr/local/share/.config/yarn/link/node_modules/.bin
  - /usr/local/libexec/lib/node_modules/npm/bin/node-gyp-bin
  - /usr/local/lib/node_modules/npm/bin/node-gyp-bin
  - /usr/local/bin/node_modules/npm/bin/node-gyp-bin
  - /usr/local/sbin
  - /usr/local/bin
  - /usr/sbin
  - /usr/bin
  - /sbin
  - /bin
  PLATFORM: linux x64
  FREEMEM: 8970514432
  TOTALMEM: 15768760320
  UPTIME: 373
  LOADAVG: 1.92578125,1.4169921875,0.6416015625
  CPUS:
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  - Intel(R) Xeon(R) CPU E5-2666 v3 @ 2.90GHz - 2900
  ENDIANNESS: LE
  VERSIONS:
  - ares: 1.15.0
  - cldr: 33.1
  - http_parser: 2.8.0
  - icu: 62.1
  - modules: 64
  - napi: 3
  - nghttp2: 1.34.0
  - node: 10.15.3
  - openssl: 1.1.0j
  - tz: 2018e
  - unicode: 11.0
  - uv: 1.23.2
  - v8: 6.8.275.32-node.51
  - zlib: 1.2.11

ERROR Summary:

  - broccoliBuilderErrorStack: Error: ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'
    at Object.realpathSync (fs.js:1456:7)
    at _walkSync (/builds/qonto/qonto-js/node_modules/broccoli-rollup/node_modules/walk-sync/index.js:101:23)
    at Function.entries (/builds/qonto/qonto-js/node_modules/broccoli-rollup/node_modules/walk-sync/index.js:34:16)
    at Object.treeFromPath (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/utils.js:81:37)
    at RollupHelper.syncInput (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/rollup-helper.js:36:39)
    at RollupHelper.build (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/rollup-helper.js:23:32)
    at BroccoliRollup.build (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/index.js:42:32)
    at resolve (/builds/qonto/qonto-js/node_modules/broccoli/lib/wrappers/transform-node.js:29:35)
    at new Promise (<anonymous>:null:null)
    at TransformNodeWrapper.build (/builds/qonto/qonto-js/node_modules/broccoli/lib/wrappers/transform-node.js:21:12)
    at Promise.resolve.then.then.then (/builds/qonto/qonto-js/node_modules/broccoli/lib/builder.js:116:28)
    at process._tickCallback (internal/process/next_tick.js:68:7)

  - code: [undefined]
  - codeFrame: ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'
  - errorMessage: ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'
        at BroccoliRollup
-~- created here: -~-
    at new Plugin (/builds/qonto/qonto-js/node_modules/broccoli-rollup/node_modules/broccoli-plugin/index.js:30:32)
    at new BroccoliRollup (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/index.js:14:9)
    at module.exports (/builds/qonto/qonto-js/ember-cli-build.js:67:23)
    at Builder.readBuildFile (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/builder.js:51:14)
    at Builder.setupBroccoliBuilder (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/builder.js:65:22)
    at new Builder (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/builder.js:31:10)
    at BuildTask.run (/builds/qonto/qonto-js/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at Promise.resolve.then (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/command.js:239:24)
    at tryCatcher (/builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:323:19)
    at invokeCallback (/builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:495:31)
    at /builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:559:14
    at flush (/builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2402:5)
    at process._tickCallback (internal/process/next_tick.js:61:11)
    at Function.Module.runMain (internal/modules/cjs/loader.js:757:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)

-~- (end) -~-
  - errorType: Build Error
  - location:
    - column: [undefined]
    - file: [undefined]
    - line: [undefined]
    - treeDir: [undefined]
  - message: ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'
        at BroccoliRollup
-~- created here: -~-
    at new Plugin (/builds/qonto/qonto-js/node_modules/broccoli-rollup/node_modules/broccoli-plugin/index.js:30:32)
    at new BroccoliRollup (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/index.js:14:9)
    at module.exports (/builds/qonto/qonto-js/ember-cli-build.js:67:23)
    at Builder.readBuildFile (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/builder.js:51:14)
    at Builder.setupBroccoliBuilder (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/builder.js:65:22)
    at new Builder (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/builder.js:31:10)
    at BuildTask.run (/builds/qonto/qonto-js/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at Promise.resolve.then (/builds/qonto/qonto-js/node_modules/ember-cli/lib/models/command.js:239:24)
    at tryCatcher (/builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:323:19)
    at invokeCallback (/builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:495:31)
    at /builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:559:14
    at flush (/builds/qonto/qonto-js/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:2402:5)
    at process._tickCallback (internal/process/next_tick.js:61:11)
    at Function.Module.runMain (internal/modules/cjs/loader.js:757:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)

-~- (end) -~-
  - name: BuildError
  - nodeAnnotation: [undefined]
  - nodeName: BroccoliRollup
  - originalErrorMessage: ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'
  - stack: Error: ENOENT: no such file or directory, lstat '/tmp/broccoli-1713V9oMBA72T6h/cache-900-broccoli_rollup/build/lib'
    at Object.realpathSync (fs.js:1456:7)
    at _walkSync (/builds/qonto/qonto-js/node_modules/broccoli-rollup/node_modules/walk-sync/index.js:101:23)
    at Function.entries (/builds/qonto/qonto-js/node_modules/broccoli-rollup/node_modules/walk-sync/index.js:34:16)
    at Object.treeFromPath (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/utils.js:81:37)
    at RollupHelper.syncInput (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/rollup-helper.js:36:39)
    at RollupHelper.build (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/rollup-helper.js:23:32)
    at BroccoliRollup.build (/builds/qonto/qonto-js/node_modules/broccoli-rollup/dist/index.js:42:32)
    at resolve (/builds/qonto/qonto-js/node_modules/broccoli/lib/wrappers/transform-node.js:29:35)
    at new Promise (<anonymous>:null:null)
    at TransformNodeWrapper.build (/builds/qonto/qonto-js/node_modules/broccoli/lib/wrappers/transform-node.js:21:12)
    at Promise.resolve.then.then.then (/builds/qonto/qonto-js/node_modules/broccoli/lib/builder.js:116:28)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

We use it roughly like this:

```js
  let fooTree = new Rollup('lib', {
    rollup: {
      input: 'foo.js',
      output: {
        file: 'assets/foo.js',
        format: 'iife',
      },
      plugins: [resolve(), commonjs()],
    },
  });
```

### PR DESCRIPTION
**Pull request by rwjblue**
Fixes #52.

### CODE REVIEW

File: tests/index.js [hunk 1]
```diff
@@ -157,6 +157,134 @@ export default index;
...
+        const output = use(createBuilder(subject));
+        // INITIAL
+        project.write({
+          'relative-input-path': {
+            'add.js': 'export default x => x + x;',
+            'index.js':
+              'import add from "./add"; const two = add(1); export default two;',
+          },
+        });
+        await output.build();
```
Comment (krisselden) [inline tests/index.js:200]: should assert the path has been restored back to the project after build

File: tests/index.js [hunk 2]
```diff
@@ -157,6 +157,134 @@ export default index;
...
+        try {
+          await output.build();
+        } catch (e) {
+          errorWasThrown = true;
+          assert.ok(
+            /Could not.*minus\.js/.test(e.message),
+            `expected error about minus.js missing but got ${e.message}`,
+          );
+        }
+        assert.ok(errorWasThrown, 'error was thrown');
```
Comment (krisselden) [inline tests/index.js:270]: after an error we should assert the pwd() is what we set it to at the start

**Comment by rwjblue** [comment]
Updated to add assertions confirming the working directory is reset after each `output.build()` invocation.

**Comment by rwjblue** [comment]
@krisselden - Mind doing another review pass?

**Pull request merged**